contracts/marketpools.js

```diff
@@ -181,14 +180,22 @@ actions.addLiquidity = async (payload) => {
     tokenPair,
     baseQuantity,
     quoteQuantity,
+    maxSlippage,
     isSignedWithActiveKey,
   } = payload;
 
   if (!api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')
-    || !api.assert(baseQuantity && api.BigNumber(baseQuantity).gt(0), 'invalid baseQuantity')
-    || !api.assert(quoteQuantity && api.BigNumber(quoteQuantity).gt(0), 'invalid quoteQuantity')
+    || !api.assert(typeof baseQuantity === 'string' && api.BigNumber(baseQuantity).gt(0), 'invalid baseQuantity')
+    || !api.assert(typeof quoteQuantity === 'string' && api.BigNumber(quoteQuantity).gt(0), 'invalid quoteQuantity')
     || !await validateTokenPair(tokenPair)) return;
 
+  let addSlippage = api.BigNumber('0.01');
+  if (maxSlippage) {
+    if (!api.assert(typeof maxSlippage === 'string'
+      && api.BigNumber(maxSlippage).gt(0) && api.BigNumber(maxSlippage).lt(50), 'maxSlippage must be greater than 0 and less than 50')) return;
```

Comment:
Do we want to allow 50?

Reply:
Without high slippage pools with low liquidity can't be used. So, may be allowing it in the contract is fine, users will make informed decision to change it to whatever % they like.

Reply:
All that will happen is they'll potentially deposit 50% less than they intended into the pool.